Assets/LeapMotion/Core/Scripts/Utils/PostProcessProvider.cs

```diff
@@ -0,0 +1,138 @@
+/******************************************************************************
+ * Copyright (C) Leap Motion, Inc. 2011-2018.                                 *
+ * Leap Motion proprietary and  confidential.                                 *
+ *                                                                            *
+ * Use subject to the terms of the Leap Motion SDK Agreement available at     *
+ * https://developer.leapmotion.com/sdk_agreement, or another agreement       *
+ * between Leap Motion and you, your company or other organization.           *
+ ******************************************************************************/
+
+using UnityEngine;
+using Leap.Unity.Attributes;
+
+namespace Leap.Unity {
+
+  public abstract class PostProcessProvider : LeapProvider {
+
+    [Tooltip("The LeapProvider whose output hand data will be copied, modified, and "
+           + "output by this post-processing provider.")]
+    [SerializeField]
+    [OnEditorChange("inputLeapProvider")]
+    protected LeapProvider _inputLeapProvider;
+    public LeapProvider inputLeapProvider {
+      get { return _inputLeapProvider; }
+      set {
+        validateInput();
+
+        if (_inputLeapProvider != null) {
+          _inputLeapProvider.OnFixedFrame -= processFixedFrame;
+          _inputLeapProvider.OnUpdateFrame -= processUpdateFrame;
+        }
+
+        _inputLeapProvider = value;
+
+        if (_inputLeapProvider != null) {
+          _inputLeapProvider.OnFixedFrame -= processFixedFrame; // safeguard double-subscription
+          _inputLeapProvider.OnFixedFrame += processFixedFrame;
+          _inputLeapProvider.OnUpdateFrame -= processUpdateFrame; // safeguard double-subscription
+          _inputLeapProvider.OnUpdateFrame += processUpdateFrame;
+        }
+      }
+    }
+
+    public enum DataUpdateMode { UpdateOnly, FixedUpdateOnly, UpdateAndFixedUpdate }
+    [Tooltip("Whether this post-processing provider should process data received from "
+           + "Update frames, FixedUpdate frames, or both. Processing both kinds of frames "
+           + "is only recommended if your post-process is stateless.")]
+    public DataUpdateMode dataUpdateMode = DataUpdateMode.UpdateOnly;
```

Comment:
>Within ProcessFrame there isn't really a way to know what kind of frame you are processing.

I discussed this with John while I was fixing the bug, there is a way to handle state correctly when consuming both events, you just use Time.inFixedTimeStep. I decided not to duplicate the functionality of Time.inFixedTimeStep and pollute the function signature with a boolean that only-matters-sometimes anyway.
https://docs.unity3d.com/ScriptReference/Time-inFixedTimeStep.html

Regarding the public field:
The example is constructed to ensure that delta times passed to the simulation step are always positive, so that the simulation doesn't break terribly. It won't be _correct_ necessarily, but it should still look feasible. (If you don't ensure positive timesteps, you get terrible breakage, which is why I put the check in there.) If changing the update mode to "both" _does_ make the simulation look much more incorrect, then I screwed something up and should address it as a bug.

Reply:
Hmmmm that makes sense, and good to know Time.inFixedTimeStep exists, I did not!  I'd still lean towards having the inertial provider be a correct implementation of a stated post process provider.  If it is supposed to serve as an example, we wouldn't want to encourage practices that could have subtle (or not subtle) errors right?